### PR TITLE
OCM-8922 | test: Automated case id:65160 Htpasswd IDP admin flow vali…

### PR DESCRIPTION
**Pull request by AkashKanni**
[OCM-8922](https://issues.redhat.com//browse/OCM-8922) | [QE] Automate OCP-65160

$ ginkgo --focus 65160 tests/e2e
Ginkgo detected a version mismatch between the Ginkgo CLI and the version of Ginkgo imported by your packages:
  Ginkgo CLI Version:
    2.11.0
  Mismatched package versions found:
    2.17.1 used by e2e

  Ginkgo will continue to attempt to run but you may see errors (including flag
  parsing errors) and should either update your go.mod or your version of the
  Ginkgo CLI to match.

  To install the matching version of the CLI run
    go install github.com/onsi/ginkgo/v2/ginkgo
  from a path that contains a go.mod file.  Alternatively you can use
    go run github.com/onsi/ginkgo/v2/ginkgo
  from a path that contains a go.mod file to invoke the matching version of the
  Ginkgo CLI.

  If you are attempting to test multiple packages that each have a different
  version of the Ginkgo library with a single Ginkgo CLI that is currently
  unsupported.
  
Running Suite: ROSA CLI e2e tests suite - /home/akanni/OCP-Repository/rosa/tests/e2e
====================================================================================
Random Seed: 1719210506

Will run 2 of 114 specs
SSSSSSSSSS••SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Ran 2 of 114 Specs in 94.964 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 112 Skipped
PASS

Ginkgo ran 1 suite in 1m36.014270667s
Test Suite Passed

### CODE REVIEW

File: tests/e2e/test_rosacli_idp.go [hunk 1]
```diff
@@ -680,4 +680,149 @@ var _ = Describe("Edit IDP",
 				Expect(textData).Should(ContainSubstring("Failed to create IDP for cluster '%s': Expected a valid certificate bundle: open %s: no such file or directory", clusterID, invalidCaFilePath))
 			})
 
+		It("can Create multiple HTPasswd IDPs including creating admin flow by the rosacli command - [id:65160]",
+			labels.Medium, labels.Runtime.Day2, labels.Feature.IDP,
```
Comment (xueli181114) [inline tests/e2e/test_rosacli_idp.go:684]: Check if `labels.Feature.IDP` is label in Describe level. No need to label in spec level.

Comment (AkashKanni) [reply tests/e2e/test_rosacli_idp.go]: Yes `labels.Feature.IDP` is available at describe level, removed it from spec level, Thanks!

File: tests/e2e/test_rosacli_idp.go [hunk 2]
```diff
@@ -680,4 +680,149 @@ var _ = Describe("Edit IDP",
...
+					Skip("IDP is not supported for external auth")
+				}
+
+				var (
+					idpNames   = []string{"htpasswdn1", "htpasswdn2"}
+					usersValue = "testuser:as65160-MSV5R-bUwmc-2506"
+					idpType    = "htpasswd"
+				)
+
+				By("Test the scenario: create admin --> create htpasswd idp with --users (htpasswdidp1) --> create htpasswd idp with --from-file (htpasswdidp2)")
```
Comment (radtriste) [inline tests/e2e/test_rosacli_idp.go:696]: Having different scenarios into one case, I would separate in several cases ...

Comment (radtriste) [reply tests/e2e/test_rosacli_idp.go:696]: Maybe one `Context` and 2 `It` ?

Comment (AkashKanni) [reply tests/e2e/test_rosacli_idp.go]: done, could you please take a look again? Thanks

File: tests/e2e/test_rosacli_idp.go [hunk 3]
```diff
@@ -680,4 +680,162 @@ var _ = Describe("Edit IDP",
 				Expect(textData).Should(ContainSubstring("Failed to create IDP for cluster '%s': Expected a valid certificate bundle: open %s: no such file or directory", clusterID, invalidCaFilePath))
 			})
 
+		Context("can Create multiple HTPasswd IDPs including creating admin flow by the rosacli command - [id:65160]",
```
Comment (radtriste) [inline tests/e2e/test_rosacli_idp.go:683]:
```suggestion
		Context("can Create multiple HTPasswd IDPs including creating admin flow - [id:65160]",
```

File: tests/e2e/test_rosacli_idp.go [hunk 4]
```diff
@@ -680,4 +680,162 @@ var _ = Describe("Edit IDP",
...
+
+					})
+
+				It("Test the scenario: create htpasswd idp with --from-file (htpasswdidp1)--> create htpasswd idp with (htpasswdidp3) --> create admin",
+					labels.Medium, labels.Runtime.Day2,
+					func() {
+
+						if profile.ClusterConfig.ExternalAuthConfig {
+							Skip("IDP is not supported for external auth")
+						}
```
Comment (radtriste) [inline tests/e2e/test_rosacli_idp.go:764]: Could be moved in a `BeforeEach` inside the `Context`, so common to both scenarios

File: tests/e2e/test_rosacli_idp.go [hunk 5]
```diff
@@ -680,4 +680,162 @@ var _ = Describe("Edit IDP",
...
+		Context("can Create multiple HTPasswd IDPs including creating admin flow by the rosacli command - [id:65160]",
+			func() {
+
+				var (
+					idpNames   = []string{"htpasswdn1", "htpasswdn2"}
+					usersValue = "testuser:as65160-MSV5R-bUwmc-2506"
+					idpType    = "htpasswd"
+				)
+
+				It("Test the scenario: create admin --> create htpasswd idp with --users (htpasswdidp1) --> create htpasswd idp with --from-file (htpasswdidp2)",
```
Comment (radtriste) [inline tests/e2e/test_rosacli_idp.go:692]:
```suggestion
				It("Create admin first",
```

File: tests/e2e/test_rosacli_idp.go [hunk 6]
```diff
@@ -680,4 +680,162 @@ var _ = Describe("Edit IDP",
...
+								idpName,
+							)
+							Expect(err).To(BeNil())
+							textData = rosaClient.Parser.TextData.Input(output).Parse().Tip()
+							Expect(textData).Should(ContainSubstring("Successfully deleted identity provider '%s' from cluster '%s'", idpName, clusterID))
+						}
+
+					})
+
+				It("Test the scenario: create htpasswd idp with --from-file (htpasswdidp1)--> create htpasswd idp with (htpasswdidp3) --> create admin",
```
Comment (radtriste) [inline tests/e2e/test_rosacli_idp.go:758]:
```suggestion
				It("Create admin last",
```

File: tests/e2e/test_rosacli_idp.go [hunk 7]
```diff
@@ -680,4 +680,162 @@ var _ = Describe("Edit IDP",
...
+						By("Create one htpasswd idp with --users flag")
+						output, err = idpService.CreateIDP(clusterID, idpNames[0],
+							"--type", idpType,
+							"--users", usersValue,
+							"-y")
+						Expect(err).To(BeNil())
+						textData = rosaClient.Parser.TextData.Input(output).Parse().Tip()
+						Expect(textData).Should(ContainSubstring("Identity Provider '%s' has been created", idpNames[0]))
+
+						By("Create one htpasswd idp with --form-file")
```
Comment (radtriste) [inline tests/e2e/test_rosacli_idp.go:715]:
```suggestion
						By("Create one htpasswd idp with --from-file")
```

File: tests/e2e/test_rosacli_idp.go [hunk 8]
```diff
@@ -680,4 +680,162 @@ var _ = Describe("Edit IDP",
...
+
+				It("Test the scenario: create htpasswd idp with --from-file (htpasswdidp1)--> create htpasswd idp with (htpasswdidp3) --> create admin",
+					labels.Medium, labels.Runtime.Day2,
+					func() {
+
+						if profile.ClusterConfig.ExternalAuthConfig {
+							Skip("IDP is not supported for external auth")
+						}
+
+						By("Create one htpasswd idp with --form-file")
```
Comment (radtriste) [inline tests/e2e/test_rosacli_idp.go:766]:
```suggestion
						By("Create one htpasswd idp with --from-file")
```

File: tests/e2e/test_rosacli_idp.go (hunk 3 above)
Comment (AkashKanni) [reply tests/e2e/test_rosacli_idp.go]: updated

File: tests/e2e/test_rosacli_idp.go (hunk 4 above)
Comment (AkashKanni) [reply tests/e2e/test_rosacli_idp.go]: done

File: tests/e2e/test_rosacli_idp.go (hunk 5 above)
Comment (AkashKanni) [reply tests/e2e/test_rosacli_idp.go]: done

File: tests/e2e/test_rosacli_idp.go (hunk 6 above)
Comment (AkashKanni) [reply tests/e2e/test_rosacli_idp.go]: done

File: tests/e2e/test_rosacli_idp.go (hunk 7 above)
Comment (AkashKanni) [reply tests/e2e/test_rosacli_idp.go]: done

File: tests/e2e/test_rosacli_idp.go (hunk 8 above)
Comment (AkashKanni) [reply tests/e2e/test_rosacli_idp.go]: corrected the typo, Thanks @radtriste!

**Review by radtriste — APPROVED** [review]
.lgtm

**Comment by AkashKanni** [comment]
@radtriste small change made, regarding adjusting expected output, /test lint it was failing due to exceeding chars in output message
could you please look once again? Thanks

**Review by radtriste — APPROVED** [review]
/lgtm

**Pull request merged**